advanced/Templates/gravity.db.sql

```diff
@@ -0,0 +1,45 @@
+CREATE TABLE whitelist (domain TEXT UNIQUE NOT NULL,  enabled BOOLEAN NOT NULL DEFAULT 1, date_added INTEGER NOT NULL DEFAULT (cast(strftime('%s', 'now') as int)), date_modified INTEGER NOT NULL DEFAULT (cast(strftime('%s', 'now') as int)), comment TEXT);
```

Comment:
Can these `UNIQUE NOT NULL` columns instead be primary keys? A primary key has these constraints asking with additional properties: https://www.w3schools.com/sql/sql_primarykey.asp

Reply:
Or, we should insert an ID column and make that the primary key.

Reply:
Yes, that's possible but conflicts with the idea of adding INTEGER ids as primary keys.
Even in the example you referenced they specify the `NOT NULL` in addition to `PRIMARY KEY`...:
```sql
CREATE TABLE Persons (
    ID int NOT NULL PRIMARY KEY,
    LastName varchar(255) NOT NULL,
    FirstName varchar(255),
    Age int
); 
```

Reply:
> Primary keys must contain UNIQUE values, and cannot contain NULL values.

Because of this, you don't need the `NOT NULL`